src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,87 @@
+            return new TreasuryIufResult(Collections.emptyList(), false);
+        }
 
-      List<Path> ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
+        if (ingestionFlowFiles != null && !ingestionFlowFiles.isEmpty()) {
+            ingestionFlowFiles.forEach(path -> {
+                File ingestionFlowFile = path.toFile();
+                log.debug("file from zip archive with name {} loaded successfully ", ingestionFlowFile.getName());
 
-     log.debug("Successfully retrieved the following files related to the ingestionFlowFileId {}: {}", ingestionFlowFileId, ingestionFlowFiles);
+                success.set(parseData(ingestionFlowFile));
 
-    } catch (Exception e) {
-      log.error("Error during TreasuryOpiIngestionActivity ingestionFlowFileId {}", ingestionFlowFileId, e);
-      return new TreasuryIufResult(Collections.emptyList(), false);
+            });
+        }
+        return new TreasuryIufResult(iufIuvList, success.get());
     }
-    return new TreasuryIufResult(iufIuvList, true);
-  }
 
+    private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileId) {
+        IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
+                .orElseThrow(() -> new IngestionFlowFileNotFoundException("Cannot found ingestionFlow having id: " + ingestionFlowFileId));
+        if (!ingestionFlowFileDTO.getFlowFileType().equals(ingestionflowFileType)) {
+            throw new IllegalArgumentException("invalid ingestionFlow file type " + ingestionFlowFileDTO.getFlowFileType());
+        }
+        return ingestionFlowFileDTO;
+    }
 
+    private List<Path> retrieveFiles(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
 
-  private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileId) {
-      IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
-              .orElseThrow(() -> new IngestionFlowFileNotFoundException("Cannot found ingestionFlow having id: " + ingestionFlowFileId));
-      if (!ingestionFlowFileDTO.getFlowFileType().equals(ingestionflowFileType)) {
-        throw new IllegalArgumentException("invalid ingestionFlow file type " + ingestionFlowFileDTO.getFlowFileType());
-      }
-    return ingestionFlowFileDTO;
-  }
+        return ingestionFlowFileRetrieverService
+                .retrieveAndUnzipFile(Path.of(ingestionFlowFileDTO.getFilePath()), ingestionFlowFileDTO.getFileName());
+    }
 
-  private  List<Path> retrieveFiles(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
+    private boolean parseData(File ingestionFlowFile) {
+
+
+        it.gov.pagopa.payhub.activities.xsd.treasury.opi14.FlussoGiornaleDiCassa flussoGiornaleDiCassa14 = null;
+        it.gov.pagopa.payhub.activities.xsd.treasury.opi161.FlussoGiornaleDiCassa flussoGiornaleDiCassa161 = null;
+
+        try {
```

Comment:
there is not any way to know what should be right unmarshaller instead of try/catching?

Reply:
It would be possible but other dependencies like DOM (Document Object Model) or XPath are required